### PR TITLE
Martin Dimitrov second task

### PR DESCRIPTION
**Pull request by MartinYDimitrov**

### CODE REVIEW

File: src/components/Footer/Index.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect, useState } from 'react';
+import styles from './Index.module.scss';
+
+function Footer() {
+  const [date, setDate] = useState('');
+
+  const timerID = setInterval(() => setDate(new Date()), 1000);
```
Comment (AV29) [inline src/components/Footer/Index.jsx:7]: I'd say here on each render new timer gets created, but you never clear it anywhere. Memory leak ?

File: src/components/Header/Index.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  const [visibility, setVisibility] = useState(false);
+
+  const clickHandler = () => {
+    setVisibility(!visibility);
+  };
+
+  return (
+    <header className={styles.header}>
+      <nav className={styles.container}>
+        <Button clickHandler={clickHandler} />
```
Comment (AV29) [inline src/components/Header/Index.jsx:16]: I suggest to use this code naming convention: 
`<Button onClick={handleClick}...`

File: src/components/Header/Index.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        {visibility && <List />}
+      </nav>
+      <div className={styles['header__title-container']}>
+        <h2 className={styles.header__title}>
+          React Treaning App
```
Comment (AV29) [inline src/components/Header/Index.jsx:21]: ...Training

File: src/components/Header/Index.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React, { useState } from 'react';
+import List from '../List/Index';
+import Button from '../Button/Index';
+import styles from './Index.module.scss';
+
+function Header() {
+  const [visibility, setVisibility] = useState(false);
```
Comment (AV29) [inline src/components/Header/Index.jsx:7]: more like isVisible and setIsVisible

File: src/components/Header/Index.jsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  const clickHandler = () => {
+    setVisibility(!visibility);
```
Comment (AV29) [inline src/components/Header/Index.jsx:10]: you can also use setVisibility(prev => !prev) syntax

File: src/App.jsx [hunk 6]
```diff
@@ -1,8 +1,27 @@
...
 class App extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = { buttonColor: 'red' };
+    this.handleCLick = this.handleCLick.bind(this);
+  }
+
+  handleCLick() {
+    this.setState((state) => {
+      const color = state.buttonColor === 'red' ? 'blue' : 'red';
```
Comment (ximet) [inline src/App.jsx:13]: really stange code: if smth == red then blue else red. I hope you can do it better

File: src/common/variables.scss [hunk 7]
```diff
@@ -0,0 +1,12 @@
+// colors
+$primary : rgba(173, 216, 230, 0.5);
+$secondary : rgba(128, 128, 128, 0.5);
+$tertiary : rgba(255, 165, 0, 0.5);
+
+
+//pixels
+$primarypixels : 100px;
```
[the remark this answers is not in the document]
Comment (ximet) [reply src/common/variables.scss:8]: or camelCase

File: src/components/Header/Index.module.css [hunk 8]
```diff
@@ -0,0 +1,26 @@
+.container {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.header {
+  display: grid;
+  grid-template-columns: 15% 1fr;
+  grid-template-rows: 100px;
```
Comment (ximet) [inline src/components/Header/Index.module.css:10]: smth in %, smth in px. I think you need to think about standards! For my opinion better use em

**Pull request closed without merging**